src/api/resolvers/tokenResolvers.js

```diff
@@ -165,29 +309,52 @@ const resolvers = {
           symbol: 'ETH',
           decimals: 18
         }
+      } else if (
+        tokenAddress === '0x89d24a6b4ccb1b6faa2625fe562bdd9a23260359'
+      ) {
+        // Fudge name and symbol for DAI v1 (SAI) to prevent confusion
+        return {
+          name: 'Sai Stablecoin',
+          symbol: 'SAI',
+          decimals: 18
+        }
       }
       const web3 = await getWeb3()
-      const contract = getTokenContract(web3, tokenAddress)
 
       try {
+        const contract = getTokenContract(web3, tokenAddress, detailedERC20ABI)
         let [name, symbol, decimals] = await Promise.all([
           contract.name().call(),
           contract.symbol().call(),
           contract.decimals().call()
         ])
-        name = web3.utils.toAscii(name)
-        symbol = web3.utils.toAscii(symbol)
         return { name, symbol, decimals }
       } catch (err) {
-        throw new Error(`Failed to get Token`)
+        try {
+          const contract = getTokenContract(
+            web3,
+            tokenAddress,
+            detailedERC20bytes32ABI
+          )
+          let [name, symbol, decimals] = await Promise.all([
+            contract.name().call(),
+            contract.symbol().call(),
+            contract.decimals().call()
+          ])
+          name = web3.utils.toAscii(name).replace(/\u0000/g, '') // eslint-disable-line no-control-regex
```

Comment:
Can we please replace these with new variables to be more descriptive with what this line is doing to the variable. Same with symbol

Reply:
sure thing.